xbmc/music/Album.cpp

```diff
@@ -52,70 +52,103 @@ CAlbum::CAlbum(const CFileItem& item)
   strAlbum = tag.GetAlbum();
   strMusicBrainzAlbumID = tag.GetMusicBrainzAlbumID();
   genre = tag.GetGenre();
-  std::vector<std::string> musicBrainAlbumArtistHints = tag.GetMusicBrainzAlbumArtistHints();
+  std::vector<std::string> musicBrainzAlbumArtistHints = tag.GetMusicBrainzAlbumArtistHints();
   strArtistDesc = tag.GetAlbumArtistString();
 
   if (!tag.GetMusicBrainzAlbumArtistID().empty())
-  { // have musicbrainz artist info, so use it
-    // Establish tag consistency - do the number of musicbrainz ids and number of names in hints or artist match
-    if (tag.GetMusicBrainzAlbumArtistID().size() != musicBrainAlbumArtistHints.size() &&
-      tag.GetMusicBrainzAlbumArtistID().size() != tag.GetAlbumArtist().size())
+  { // Have musicbrainz artist info, so use it
+    
+    // Vector of possible separators in the order least likely to be part of artist name
+    const char *sepinit[] = { " feat. ", ";", ":", "|", "#", "/", ",", "&" };
+    std::vector<std::string> separators(sepinit, std::end(sepinit));
```

Comment:
Is sepinit used only to initialize the vector? in c++11 you should be able to do
`std::vector<std::string> separators{ " feat. ", ";", ":", "|", "#", "/", ",", "&" };`
Same thing in Song.cpp